tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/metrics/BulkheadMetricBean.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.fault.tolerance.tck.metrics;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.faulttolerance.Asynchronous;
+import org.eclipse.microprofile.faulttolerance.Bulkhead;
+
+@ApplicationScoped
+public class BulkheadMetricBean {
+
+    @Bulkhead(2)
+    public void waitFor(long milliseconds) {
```

Comment:
In tests, it's better to control the length of execution. Unnecessary sleeps make the tests unreliable and long to run. So it's better to pass a future instead of timeout to have complete control over the timing, such as:

```
public void waitFor(Future condition) {
    condition.get();
}
```

and call it with:

```
CompletableFuture cf = new CompletableFuture();
waitFor(cf);
// do something that should happen meanwhile
cf.complete();
```

Reply:
Ah, that's a much better way of controlling the execution order. I'll update the test to use this technique.